camel-core/src/main/java/org/apache/camel/component/xslt/XsltEndpoint.java

```diff
@@ -71,6 +72,8 @@
     @UriParam
     private boolean saxon;
     @UriParam(label = "advanced")
+    private List<Object> saxonExtensionFunctions;
```

Comment:
You need to add some logic to xstl component so people can provide the functions in the ur, as a comma separated value of names to lookup in the registry, eg `saxonExtensionFunctions=#foo,#bar` etc. to lookup foo and bar from the registry. eg add an unit test in camel-spring to see how easy/hard it is to use as a spring/blueprint end user.